### PR TITLE
8-sided Bridge with DKG Deployment

### PR DESCRIPTION
**Pull request by AtelyPham**
## Summary of changes
_Provide a detailed description of proposed changes._
- Add 2 new chains: Avalanche & Scroll
- Add 8-sided bridge with dkg deployment

### Proposed area of change
_Put an `x` in the boxes that apply._

- [x] `apps/bridge-dapp`
- [ ] `apps/stats-dapp`
- [ ] `apps/webbsite`
- [ ] `libs/webb-ui-components`

### Reference issue to close (if applicable)

- Closes https://github.com/webb-tools/webb-dapp/issues/1044

### CODE REVIEW

File: libs/dapp-config/src/anchors/anchor-config.ts [hunk 1]
```diff
@@ -10,15 +10,19 @@ import { calculateTypedChainId, ChainType } from '@webb-tools/sdk-core';
...
-      '0xa1a2b7e08793b3033122b83cbee56726678588b5',
+      '0x12f721c568e907126e13d6664ebec606ee65ed3c',
     [PresetTypedChainId.MoonbaseAlpha]:
-      '0xa1a2b7e08793b3033122b83cbee56726678588b5',
+      '0x12f721c568e907126e13d6664ebec606ee65ed3c',
+    [PresetTypedChainId.AvalancheFuji]:
+      '0x12f721c568e907126e13d6664ebec606ee65ed3c',
+    [PresetTypedChainId.ScrollAlpha]:
+      '0x12f721c568e907126e13d6664ebec606ee65ed3c',
   },
```
Comment (dutterbutter) [inline libs/dapp-config/src/anchors/anchor-config.ts:26]: @AtelyPham we do not want to replace these addresses rather we want to include the. Specifically, `0xa1a2b7e08793b3033122b83cbee56726678588b5` is the anchor address for the mocked webb bridge and `0x12f721c568e907126e13d6664ebec606ee65ed3c` is another webb based bridge. Collectively, the dApp would then support two distinct bridges with two distinct webb bridged assets.

Comment (AtelyPham) [reply libs/dapp-config/src/anchors/anchor-config.ts:26]: Okay, so I'll need to update the logic to be able to select multiple bridges. Do we have the design for that? @dutterbutter

**Comment by dutterbutter** [comment]
@salman01zp  Likewise for this PR we need relayer4 configured with these chains / bridge in order to test before moving to prod.

**Comment by dutterbutter** [comment]
The balance on withdraw seems to be an issue. I deposited .01 from sepolia to mumbai and wrapped into webbStandalone. Yet on the withdraw component, it indicates I have insufficient balance. I have confirmed I am connected to Mumbai and the notes table displays my deposit correctly. 

In addition, keeps asking me to "switch chain to withdraw" despite being connected to Mumbai. When selected it changes webbStandalone to webbWETH. @AtelyPham can you attempt to reproduce. 

tinpr

**Comment by AtelyPham** [comment]
Fixed the issue, but we need to create an additional task for getting the balances from the notes for all available fungible currencies. cc @dutterbutter

https://user-images.githubusercontent.com/60747384/227602017-142fe713-2e13-40f2-8eb2-2bcdd78d876a.mp4

**Comment by dutterbutter** [comment]
@salman01zp doesn't seem the relayer correctly relays the transactions. Please try to reproduce. It seems to get stuck at sending tx to relayer...

**Comment by dutterbutter** [comment]
Dependent on https://github.com/webb-tools/relayer/issues/401

**Comment by shekohex** [comment]
> Dependent on [webb-tools/relayer#401](https://github.com/webb-tools/relayer/issues/401)

Issue resolved and PR got merged.

**Comment by dutterbutter** [comment]
@AtelyPham please resolve conflicts and then we can finalize the review here.

**Comment by dutterbutter** [comment]
Can't seem to fetch fee info. 

CleanShot 2023-03-28 at 09 59 11@2x

**Comment by shekohex** [comment]
> Cannot withdraw now because the relayer failed to validate the fee.

Bug reported here too: https://github.com/webb-tools/relayer/issues/404

**Comment by shekohex** [comment]
Both https://github.com/webb-tools/relayer/issues/404 and https://github.com/webb-tools/relayer/issues/406 got solved today!

**Comment by dutterbutter** [comment]
@AtelyPham the relayers should be updated now. Please address conflicts.

**Comment by dutterbutter** [comment]
With https://github.com/webb-tools/webb-dapp/pull/1088 merged, this should be resolved. Seems to fine for transfer but not for withdraw. @AtelyPham can you investigate / reproduce.

https://user-images.githubusercontent.com/29983536/229169975-45305d3c-b7fd-4780-8b5e-25e2f0461965.mp4

**Comment by dutterbutter** [comment]
Please update deployment addresses to reflect the latest deployments outlined [here](https://www.notion.so/hicommonwealth/Bridge-Deployments-54496ac2c29d4a7f9867bb9fa3ed1b2a#908560a09a704838931f61b87ae070f7).

File: libs/dapp-config/src/anchors/anchor-config.ts [hunk 2]
```diff
@@ -1,38 +1,54 @@
-import { CurrencyRole, EVMChainId } from '@webb-tools/dapp-types';
-import { ChainType, calculateTypedChainId } from '@webb-tools/sdk-core';
+import { EVMChainId } from '@webb-tools/dapp-types';
+import { calculateTypedChainId, ChainType } from '@webb-tools/sdk-core';
 
-import { ApiConfig } from '../api-config';
-import { CurrencyConfig } from '../currencies';
+// 0xa1a2b7e08793b3033122b83cbee56726678588b5 - webbWETH - mocked backend
+// 0x38e7aa90c77f86747fab355eecaa0c2e4c3a463d - webbAlpha - mocked backend
+// 0x9678647b9fcb0039652a16dba688bd067d6e5077 - webbStandalone - dkg backend
```
Comment (dutterbutter) [inline libs/dapp-config/src/anchors/anchor-config.ts:6]: We can remove reference to `0x9678647b9fcb0039652a16dba688bd067d6e5077 - webbStandalone - dkg backend` in favour of `0xf8c9d24e3bc3e2d3eddde507079b08e82f239fc4 - webbtTNT-standalone`

Comment (dutterbutter) [reply libs/dapp-config/src/anchors/anchor-config.ts:6]: No need to support both

**Comment by dutterbutter** [comment]
Please resolve conflicts @AtelyPham

**Comment by AtelyPham** [comment]
It seems the issue is related to the Metamask updates, you might want to try removing the network from your Metamask Settings in the network tab and re-adding it from the dapp. Doing so helped me resolve the issue.

I also included two additional commits for parsing and displaying errors more clearly. Additionally, I extracted the switch and chain login into a single method.

**Comment by dutterbutter** [comment]
Caused a dApp crash when changing the value of the MAX refund. 

Steps to reproduce:
1. Make deposit where destination chain is Polygon using `webbStandalone` 
2. Switch to Polygon on Withdraw tab
3. Fill in the corresponding withdraw details
4. Checkbox refund
5. Select MAX for refund
6. Change the value 

CleanShot 2023-04-14 at 15 14 49@2x

**Comment by dutterbutter** [comment]
Please address overflow issue in UI: 

CleanShot 2023-04-14 at 15 18 58@2x

**Comment by AtelyPham** [comment]
It seems the DKG backend has crashed (the fungible token is `webbStandalone`), and the proposal has failed to be signed. I need further debugging with the team 😢.

**Comment by salman01zp** [comment]
In the case of DKG getting crashed, we lose contract ownership. Therefore we will need another deployment and transfer ownership to the current DKG.
Current DKG status: `STALLED`

**Comment by shekohex** [comment]
@AtelyPham @dutterbutter @salman01zp I guess we better now switch our development deployment to our Testnets, we can iterate faster there and anyone could create a deployment easily with one click. 

Just for reference, here are the tasks that I'm working on right now to achieve the above said goal:

- [ ] https://github.com/webb-tools/orbit/issues/17
- [ ] https://github.com/webb-tools/relayer/issues/464

**Comment by AtelyPham** [comment]
Added new Orbit chains, and anchor deployments, removed deprecated network. Next, we need `webbtTNT` to test the bridge.

Resolved: https://github.com/webb-tools/webb-dapp/issues/1109

File: libs/dapp-config/src/chains/evm/index.tsx [hunk 3]
```diff
@@ -175,4 +143,83 @@ export const chainsConfig: Record<number, ChainConfig> = {
...
+    chainType: ChainType.EVM,
+    group: 'eth',
+    chainId: EVMChainId.AthenaOrbit,
+    name: 'Athena Orbit',
+    base: 'webb-dev',
+    tag: 'test',
+    url: 'https://athena-testnet.webb.tools',
+    evmRpcUrls: ['https://athena-testnet.webb.tools'],
+    blockExplorerStub: 'https://athena-explorer.webb.tools',
+    logo: GanacheLogo,
```
Comment (shekohex) [inline libs/dapp-config/src/chains/evm/index.tsx:171]: I guess we need to use the new logo here?

Comment (AtelyPham) [reply libs/dapp-config/src/chains/evm/index.tsx:171]: The logo here is deprecated, we are using the dynamic `ChainIcon` component to render the chain logo based on the chain name.

**Comment by AtelyPham** [comment]
### Resolved items after the e2e testing with the team.

- [x]  Ensure handling allowance approval correctly
- [x]  Update transfer flow to include the relayer fee
- [x]  Fix recipient address UI overflow
- [x]  Verify fee should be extracted from the note or user account
- [x]  The dapp hangs when the relayer responds to the network error

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**